### ISSUE TITLE
PowerShell breaks Perl6's (Rakudo's) command-line editing on macOS

### ISSUE THREAD

**Issue by mklement0** [PowerShell/PowerShell#7920]
Observed in both Terminal.app and iTerm2.

Once you invoke PowerShell in a given Bash session, command-line editing in a subsequent invocation of the Perl6 REPL stops working.

Even just invoking PowerShell with `-command`, without entering an interactive session, produces the symptom.

With PowerShell defined as the default shell, using command-line editing in the Perl6 REPL is unavailable altogether.

Steps to reproduce
------------------

On macOS: 

* First, install Perl6 with `brew install perl6`.
* Run `perl6` to enter the Perl6 REPL and submit a command such as `'hi'` - verify that it can be recalled by pressing the up-arrow.
* Exit the REPL with `exit`
* Run a PowerShell command such as `pwsh -noprofile -command get-date`
* Enter the Perl6 REPL again with `perl6`
* Try to recall the previous command by pressing the up-arrow.

Expected behavior
-----------------

The most recently submitted command should be recalled from the command history.

Actual behavior
---------------

Nothing happens.

Even using the arrow keys to navigate within the current line isn't working.

Environment data
----------------

```powershell
PowerShell Core v6.1.0 on macOS 10.13.6
```

**Comment by BrucePay** [comment]
@mklement0 Why is this not a Perl 6 bug?  It sounds like it's failing to properly (re)initialize the tty on startup.

**Comment by mklement0** [comment]
@BrucePay: 

Since it doesn't happen with any of the major POSIX-like shells (`bash`, `dash`, `ksh`, `zsh`), I thought I'd start with PowerShell.

The fact that even a _noninteractive_, `-Command`-only invocation of PowerShell breaks subsequent command-line editing in Perl6 is especially curious.

Note that the editing isn't fully broken, since you can still type and submit commands; however, the arrow keys stop working.

**Comment by BrucePay** [comment]
@mklement0 

> The fact that even a noninteractive, -Command-only invocation of PowerShell breaks subsequent command-line editing in Perl6 is especially curious.

Curious indeed. But it still sounds like a Perl 6 bug. It shouldn't matter what we did when we were running. After we're done, Perl has control and it's not initializing things properly. Does running PowerShell before running any of the other shells cause their editing to break? No? Sounds like a Perl 6 bug.

**Comment by mklement0** [comment]
Good point, @BrucePay, but before I create an issue in the Perl6 repo: do you have any pointers as to what (a) PowerShell may be doing differently than other shells, and (b) what Perl6 may be neglecting to do to cause this behavior?
